core/types/rollup_cost.go

```diff
@@ -283,3 +336,93 @@ func l1CostHelper(gasWithOverhead, l1BaseFee, scalar *big.Int) *big.Int {
 	fee.Mul(fee, l1BaseFee).Mul(fee, scalar).Div(fee, oneMillion)
 	return fee
 }
+
+func extractEcotoneFeeParams(l1FeeParams []byte) (l1BaseFeeScalar, l1BlobBaseFeeScalar *big.Int) {
+	offset := scalarSectionStart
+	l1BaseFeeScalar = new(big.Int).SetBytes(l1FeeParams[offset : offset+4])
+	l1BlobBaseFeeScalar = new(big.Int).SetBytes(l1FeeParams[offset+4 : offset+8])
+	return
+}
+
+func bedrockCalldataGasUsed(costData RollupCostData) (calldataGasUsed *big.Int) {
+	calldataGas := (costData.zeroes * params.TxDataZeroGas) + (costData.ones * params.TxDataNonZeroGasEIP2028)
+	return new(big.Int).SetUint64(calldataGas)
+}
+
+// FlzCompressLen returns the length of the data after compression through FastLZ, based on
+// https://github.com/Vectorized/solady/blob/5315d937d79b335c668896d7533ac603adac5315/js/solady.js
+func FlzCompressLen(ib []byte) uint32 {
```

Comment:
can we have a fuzz test for this?

Reply:
I'll add this to the monorepo PR w/ the differential test that @sebastianst mentioned above.

Reply:
Don't feel so strongly about it but since the code is here shouldn't the fuzz test be here too?   We should fuzz test the solidity implementation in the monorepo I think, then differential test would be there too.